lib/IRGen/MetadataPath.h

```diff
@@ -186,10 +186,20 @@ class MetadataPath {
     for (auto C : Path) {
       switch (C.getKind()) {
       case Component::Kind::NominalParent:
+        // Work around MSVC error: syntax error 'template'
+#if defined(_MSC_VER)
+        Root = A.createParent(Root);
+#else
         Root = A.template createParent(Root);
+#endif
```

Comment:
The `template` shouldn't be required by Clang, either, since no explicit template arguments are being given. We should be able to unconditionally remove it from both of these places.